src/ipyradiant/visualization/interactive_exploration.py

```diff
@@ -0,0 +1,309 @@
+        button = W.Button(
+            description="Remove Temporary Nodes",
+            layout=W.Layout(width="25%", height="40px"),
+            disabled=False,
+        )
+        button.on_click(self.remove_temp_nodes)
+        return button
+
+    @trt.default("selected_node")
+    def _create_default_selected_node(self):
+        return None
+
+    @trt.default("cyto_style")
+    def _create_cyto_style(self):
+        return DEFAULT_CYTO_STYLE
+
+    @trt.default("rdf_graph")
+    def _create_rdf_graph(self):
+        return rdflib.Graph()
+
+    @trt.default("cyto_graph")
+    def _create_cyto_graph(self):
+        return ipycytoscape.CytoscapeWidget()
+
+    @trt.default("layout")
+    def _create_layout(self):
+        return W.Layout(width="80%")
+
+    @trt.observe("cyto_graph")
+    def update_cyto_graph(self, change):
+        self.cyto_graph.set_layout(name=self.cyto_graph_layout)
+        self.cyto_graph.set_style(self.cyto_style)
+        # on is a callback for cyto_graph instance (must be set on each instance)
+        self.cyto_graph.on("node", "click", self.log_node_clicks)
+        # Here we have to set the children again so that the changes propogate to the front end
+        # automatically. Ideally this would be done with traits but did not seem to work. LINK TO GITHUB ISSUE:
+        # https://github.com/jupyrdf/ipyradiant/issues/79
+        self.children = (
+            self.cyto_graph,
+            W.HBox(
+                children=[
+                    self.expand_button,
+                    self.undo_button,
+                    self.remove_temp_nodes_button,
+                ]
+            ),
+        )
+
+    @trt.validate("children")
+    def validate_children(self, proposal):
+        children = proposal.value
+        if not children:
+            children = (
+                self.cyto_graph,
+                W.HBox(
+                    children=[
+                        self.expand_button,
+                        self.undo_button,
+                        self.remove_temp_nodes_button,
+                    ]
+                ),
+            )
+        return children
+
+    def get_node(self, node: Node) -> Node:
+        """
+        This function is used to find a node given the id of a node copy.
+        """
+
+        for node_obj in self.cyto_graph.graph.nodes:
+            if node_obj.data["id"] == node["data"]["id"]:
+                return node_obj
+        # TODO: Make this function return None and log a warning if not node not found.
+        raise ValueError("Node not found in cytoscape.graph.nodes.")
+
+    def log_node_clicks(self, node: Node):
+        """
+        This function works with registering a click on a node. This will mark the node as selected and change the color of the
+        selected node.
+        """
+
+        node_object = self.get_node(node)
+
+        if self.selected_node == node_object:
+            node_object.classes = "clicked"
```

Comment:
Not sure about this, but if `classes` can have multiple ones applied, this may override other classes on the node_object.  This may be an issue in the future...

```python
node_classes = {cls.strip() for node_object.classes.split()}
if "clicked" not in node_classes:
    node_object.classes += " clicked"
```

Reply:
node_object -> cyto_node (and static type)